### PR TITLE
Block options and block location setting

### PR DESCRIPTION
**Pull request by John-Joubert**
Gareth - these options are working.
You can now choose one, none, or all of the blocks for display.
You can now choose to place the blocks on the left or right side.
-- John

### CODE REVIEW

**Comment by badmechanic** [comment]
Oh!  So I did get it !

Please have a look at what I sent.

LOL - that travis filter looks worse than lint!

John

________________________________
From: John Joubert ***@***.***>
Sent: Friday, April 30, 2021 5:00 AM
To: gjb2048/moodle-format_topcoll ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [gjb2048/moodle-format_topcoll] Block options and block location setting (#98)


Gareth - these options are working.
You can now choose one, none, or all of the blocks for display.
You can now choose to place the blocks on the left or right side.
-- John

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/gjb2048/moodle-format_topcoll/pull/98<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213940669%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=A%2BsiQsTJ5ikSRDmN2bgka0vfdomjWEhw7Kd%2BDUbAqEE%3D&reserved=0>

Commit Summary

  *   Block options and block location setting

File Changes

  *   M lang/en/format_topcoll.php<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98%2Ffiles%23diff-972c351ef4188f1f82cc82207e98147ccbd23e43ddf222a3e5c3f8df5e475d04&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213950663%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=0q2LdGrz6yIl0vyYe6TNzee0vJbKxO6MvOZGo%2FF0dYU%3D&reserved=0> (8)
  *   M lang/en_us/format_topcoll.php<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98%2Ffiles%23diff-c0088f30815fb5f4137495566dc9e7b9695331b1d54e3338fb13677ab4c55e85&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213960658%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=pa7EN5HpvN%2FsjWXmRjwETU6iNJUJ1K9dCXg%2BHcr3nuY%3D&reserved=0> (7)
  *   M lib.php<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98%2Ffiles%23diff-8f5ff72ef1cf59199cfc4bbf16f2475d09e448601242f0d2c51bb94bd4aa1395&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213960658%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ztRgANOQFQ27p3afeGeXYx97Q3YFLLkwuOv9WzGWdaE%3D&reserved=0> (45)
  *   M settings.php<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98%2Ffiles%23diff-4092c491c25e7e19aceafbe3c49d234ce7d1eca769519197d2ccba7bde75a27e&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213970652%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yCHFQumnHHeynSL2RrOFJwPOHPH84ZscOvwswEZFj94%3D&reserved=0> (20)

Patch Links:

  *   https://github.com/gjb2048/moodle-format_topcoll/pull/98.patch<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98.patch&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213970652%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ur3vmh480ijcDWRkgyp01EpPF6uDcgbQXS4SLcH7NCc%3D&reserved=0>
  *   https://github.com/gjb2048/moodle-format_topcoll/pull/98.diff<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgjb2048%2Fmoodle-format_topcoll%2Fpull%2F98.diff&data=04%7C01%7C%7Ce80637459a884a23ceea08d90b94da90%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637553556213970652%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=WBj9mjecAecH9QVU7mEHnZ8SCLg2bbToM7nBiGhY%2FG8%3D&reserved=0>

**Comment by gjb2048** [comment]
@John-Joubert Thanks, will look as soon as I can.

File: settings.php [hunk 1]
```diff
@@ -48,6 +48,26 @@
     );
     $settings->add(new admin_setting_configselect($name, $title, $description, $default, $choices));
 
+    /* Toggle display block choices */
+    $name = 'format_topcoll/defaultdisplayblocks';
+    $title = get_string('defaultdisplayblocks', 'format_topcoll');
+    $description = get_string('defaultdisplayblocks_desc', 'format_topcoll');
+    $default = array('0,1,2,3');
+    $choices = array('search_forums','news_items','calendar_upcoming','recent_activity');
```
Comment (gjb2048) [inline settings.php]: Not quite!  Use an associative array -> array(
'search_forums' => get_string(...),
'news_items' => get_string(...),
'calendar_upcoming' => get_string(...),
'recent_activity' => get_string(...)
)

then need to work out how to set the default.

Comment (gjb2048) [reply settings.php]: Also this list needs to be generated from the list of blocks installed.  As the above are 'core' blocks then should not be an issue with the default.

File: settings.php [hunk 2]
```diff
@@ -48,6 +48,26 @@
...
+    $settings->add(new admin_setting_configmultiselect($name, $title, $description, $default, $choices));
+    
+    /*  Toggle blocks location. */
```
Comment (gjb2048) [inline settings.php]: Please use single inline comments for one line '//' or only one block comment '/* .... */'

File: lib.php [hunk 3]
```diff
@@ -339,12 +339,49 @@ public function ajax_section_move() {
...
+     *
      */
     public function get_default_blocks() {
-        return array(
-            BLOCK_POS_LEFT => array(),
-            BLOCK_POS_RIGHT => array('search_forums', 'news_items', 'calendar_upcoming', 'recent_activity')
-        );
+        $blocklist=array();
+        $use_def_blocks = explode (",",get_config('format_topcoll', 'defaultdisplayblocks'));
+        foreach($use_def_blocks as $blockindex)
```
Comment (gjb2048) [inline lib.php]: This won't be needed as may just need an array_keys thing on the 'explode' need to debug when you change the setting.

File: lib.php [hunk 4]
```diff
@@ -339,12 +339,49 @@ public function ajax_section_move() {
...
+        }
+        /* Figure out which side it goes on, and put it there */
+        $def_blocks_loc = get_config('format_topcoll','defaultdisplayblocksloc');
+        if($def_blocks_loc == 1) /* Right Side */
+        {
+            return array( 
+                BLOCK_POS_LEFT => array(),
+                BLOCK_POS_RIGHT => $blocklist 
+            );
+        }
```
Comment (gjb2048) [inline lib.php:358]: Need to follow the Moodle coding standard so '} else {' on one line.

File: lib.php [hunk 5]
```diff
@@ -339,12 +339,49 @@ public function ajax_section_move() {
...
+                    array_push($blocklist,'calendar_upcoming'); 
+                    break;
+                case '3':
+                    array_push($blocklist,'recent_activity');
+                    break;
+                default:
+            }
+        }
+        /* Figure out which side it goes on, and put it there */
+        $def_blocks_loc = get_config('format_topcoll','defaultdisplayblocksloc');
```
Comment (gjb2048) [inline lib.php]: Combine these two lines as $def_blocks_loc not needed as only used once.  I also think this could be simpler.

File: lib.php [hunk 6]
```diff
@@ -339,12 +339,49 @@ public function ajax_section_move() {
...
+            );
+        }
+        else                     /* Left Side  */
+        {
+            return array( 
+                BLOCK_POS_LEFT => $blocklist,
+                BLOCK_POS_RIGHT => array()
+            );
+        } 
     }
```
Comment (gjb2048) [inline lib.php:365]: How about:


```
if (get_config('format_topcoll','defaultdisplayblocksloc') == 1) {
   $bpl = array();
   $bpr = $blocklist;
} else {
   $bpl = $blocklist;
   $bpr = array();
}

return array( 
    BLOCK_POS_LEFT => $bpl,
    BLOCK_POS_RIGHT => $bpr
);
```

File: lang/en_us/format_topcoll.php [hunk 7]
```diff
@@ -67,3 +67,10 @@
 // Capabilities.
 $string['topcoll:changecolour'] = 'Change or reset the color';
 
+// Toggle Display Blocks
```
Comment (gjb2048) [inline lang/en_us/format_topcoll.php]: If the language strings are identical to 'en' then this is not needed.

**Comment by John-Joubert** [comment]
I like it -
one return point from the routine, instead of two.

I'll add it.

________________________________

**Comment by badmechanic** [comment]
The problem here is that the returned value is a single string, not an array.
The value returned for the first block, and the last block would be:
"0,3" - single string that isn't very useful without taking apart the string.
Thus, the explode routine.
I've spent some time with some searching to see if there is a way to automatically refer to a csv string as an array - and I've not found anything.
It seems that explode is the common method.

John

________________________________

**Comment by badmechanic** [comment]
OK Fixed

________________________________

**Comment by badmechanic** [comment]
So are you suggesting a shift from the standard list of four blocks that you have used previously in your code, to a larger list of all available blocks?

John

________________________________

**Comment by badmechanic** [comment]
Done

________________________________

**Comment by gjb2048** [comment]
@badmechanic Ok, see:

- https://www.php.net/manual/en/function.explode.php
- Also don't have 0, 1, 2 as the keys, but the default block names (not their lang string names) and the lang strings as the values in the array passed to the setting i.e. the list of installed blocks.  The default var will then be the defaults as before.
- Then the string you get when interrogating the setting will be the comma separated list of the key values selected which you can then use 'explode' on to convert to an array.

**Comment by badmechanic** [comment]
OK - I understand what you're saying about the blocks using that kind of array, I'm hoping that the multichoice can handle that.  I suppose that since you're telling me this, that there is a very strong chance that it does. LOL.

🙂

________________________________

**Comment by John-Joubert** [comment]
So, I don't have the name change in yet -
but this is what lib.php looks like now.

As you can see I tightend up the foreach block.
I also have all ohter changes you asked for - except for changing the way the multiselector works in settings.php.
I'm going to start on that now.
Then I'll come back here and fix the code below to use the new multiselect from settings.php.



    public function get_default_blocks() {

        // Build the array of strings of the list of blocks to be populated
        $use_def_blocks = explode (",",get_config('format_topcoll', 'defaultdisplayblocks'));
        $blocklist_arr=array(0=>"search_forums", 1=>"news_items", 2=>"calendar_upcoming", 3=>"recent_activity");
        $blocklist=array();
        foreach($use_def_blocks as $blockindex) {
            array_push($blocklist,$blocklist_arr[$blockindex]);
        }

        // Assign the location side for the blocks. defaultdisplayblocksloc: 1=right, 2=left
        if (get_config('format_topcoll','defaultdisplayblocksloc') == 1) {
            $bpr = $blocklist;
            $bpl = array();
        } else {
            $bpr = array();
            $bpl = $blocklist;
        }

        // return our block list on the correct side
        return array(
            BLOCK_POS_RIGHT => $bpr,
            BLOCK_POS_LEFT  => $bpl
        );
    }


________________________________

**Comment by John-Joubert** [comment]
OK - I think it's all working as you want it to.
I'm hoping you an see this latest commit and push to noblocks_option.
Message of the commit is "New fixes to noblocks_option branch as per gjb2048". 
Thanks for all kind help.
Let me know if I need to change anything.

**Comment by badmechanic** [comment]
Allright - I'm sending in a commit.
Branch name is "noblocks_options"
Commit title is "it's all working now" (terrible name for a commit".tell

I don't see anything in the tool to say to "pull" anything - I guess I'm still using the one from before.
Not too sure.

Let me know if you see my new code to review.
I'm hoping I did it correctly. -
If I was supposed to push a whole new branch off of "noblocks_optoins"
I could have sent that instead - as I had created a branch.
And then later merged those changes into noblocks_options - so i could just send you a single commit.
Let me know if this was the wrong method.

Here's the new code in settings.php:
-----------------------------------------------------------------------------
    /* Toggle display block choices */
    $name = 'format_topcoll/defaultdisplayblocks';
    $title = get_string('defaultdisplayblocks', 'format_topcoll');
    $description = get_string('defaultdisplayblocks_desc', 'format_topcoll');
    $default = array('search_forums','news_items','calendar_upcoming','recent_activity');
    $choices = array('search_forums'=>'Search forums','news_items'=>'Latest announcements',
                     'calendar_upcoming'=>'Upcoming events','recent_activity'=>'Recent activity');
    $settings->add(new admin_setting_configmultiselect($name, $title, $description, $default, $choices));

    /*  Toggle blocks location. 1 = right, 2 = left */
    $name = 'format_topcoll/defaultdisplayblocksloc';
    $title = get_string('defaultdisplayblocksloc', 'format_topcoll');
    $description = get_string('defaultdisplayblocksloc_desc', 'format_topcoll');
    $default = 1;
    $choices = array(
        1 => new lang_string('right', 'format_topcoll'),   // Right.
        2 => new lang_string('left', 'format_topcoll'),    // Left.
    );
    $settings->add(new admin_setting_configselect($name, $title, $description, $default, $choices));

Here's the new code in lib.php.
----------------------------------------------------------------------------------
    public function get_default_blocks() {

        // Assign the location side for the blocks. defaultdisplayblocksloc: 1=right, 2=left
        // Then put the string list of blocks on the side location
        $blocklist=explode(',',get_config('format_topcoll','defaultdisplayblocks'));
        if (get_config('format_topcoll','defaultdisplayblocksloc') == 1) {
            $bpr = $blocklist;
            $bpl = array();
        } else {
            $bpr = array();
            $bpl = $blocklist;
        }

        // return our block list on the correct side
        return array(
            BLOCK_POS_RIGHT => $bpr,
            BLOCK_POS_LEFT  => $bpl
        );
    }


What's great is that you were absolutely correct in your request (but you knew that already).
It made a much tighter, and simpler piece of code.

Thank you!



________________________________

File: settings.php [hunk 8]
```diff
@@ -48,6 +48,26 @@
     );
     $settings->add(new admin_setting_configselect($name, $title, $description, $default, $choices));
 
+    /* Toggle display block choices */
+    $name = 'format_topcoll/defaultdisplayblocks';
+    $title = get_string('defaultdisplayblocks', 'format_topcoll');
+    $description = get_string('defaultdisplayblocks_desc', 'format_topcoll');
+    $default = array('search_forums','news_items','calendar_upcoming','recent_activity');
+    $choices = array('search_forums'=>'Search forums','news_items'=>'Latest announcements',
```
Comment (gjb2048) [inline settings.php]: This needs to be a list of installed blocks from the block manager with 'get_strings' on the other end, i.e get_string('pluginname', 'block_recent_activity')

**Comment by gjb2048** [comment]
Please see https://github.com/gjb2048/moodle-format_topcoll/pull/98/files#r627627923

**Comment by gjb2048** [comment]
Errors need fixing too: https://travis-ci.org/github/gjb2048/moodle-format_topcoll/jobs/769629900

File: settings.php (hunk 8 above)
Comment (gjb2048) [reply settings.php]: i.e. as stated before:

Use an associative array -> array(
'search_forums' => get_string(...),
'news_items' => get_string(...),
'calendar_upcoming' => get_string(...),
'recent_activity' => get_string(...)
)

but the keys will be from the list of installed blocks.

Comment (gjb2048) [reply settings.php]: i.e. see what core_plugin_manager::instance()->get_enabled_plugins('block') returns.

Comment (gjb2048) [reply settings.php]: or if that has agro, then core_plugin_manager::instance()->get_plugins_of_type('block').

**Comment by gjb2048** [comment]
Also.... if any of the defaults are not in the choices, then they should be removed from the defaults.

**Comment by badmechanic** [comment]
Really?
I'm surprised - because it hasn't affected the run time.  I thought that defaults were only used when nothing else was ever set before.
Once they were set, then the next time you look at the selector, it's information always reflects stored information - not default information.
Don't you think that's why information in default is separate from stored?

I mean, it's easy enough to change - but it seems to be redundant, or actually - not even needed once a system has stored values from the settings.
At that point - the default array could be empty, as it won't be used.

Or - are what you going for is cosmetics? You'd like to see the list of chosen blocks on the bottom under the box listed as "defaults"?
I'm trying to think how this will work the first time we call it.  Will the retrieve of that value from the settings.php even work before that data is stored?
Is there a return value I can count on to know it's not been called, and then set the defaults to all of them?
I'll have to give this some thought.

________________________________

**Comment by John-Joubert** [comment]
Ah OK- I can do that.  🙂

________________________________

**Comment by John-Joubert** [comment]
Good pointer - thank you!

________________________________

**Comment by gjb2048** [comment]
@badmechanic I'm not really a 'cosmetics' person in this context.  So... take the possible scenario(s) that CT is installed where one of the default blocks has been removed / deactivated, then as no value is set then this is a bad state to set the default as one of the blocks that can't be used.

**Comment by badmechanic** [comment]
Oh yes! I see that. Gotcha.

________________________________

**Comment by John-Joubert** [comment]
OK Done.  Travis is tricky.
I think lint is easier!

________________________________

**Comment by John-Joubert** [comment]
OK - so this is what this returns:

die(print_r(core_plugin_manager::instance()->get_enabled_plugins('block')));

print_r looks like this:
Array ( [activity_modules] => activity_modules [activity_results] => activity_results [admin_bookmarks] => admin_bookmarks [badges] => badges [blog_menu] => blog_menu [blog_recent] => blog_recent [blog_tags] => blog_tags [calendar_month] => calendar_month [calendar_upcoming] => calendar_upcoming [comments] => comments [completionstatus] => completionstatus [course_list] => course_list [course_summary] => course_summary [feedback] => feedback [globalsearch] => globalsearch [glossary_random] => glossary_random [html] => html [login] => login [lp] => lp [mentees] => mentees [mnet_hosts] => mnet_hosts [myoverview] => myoverview [myprofile] => myprofile [navigation] => navigation [news_items] => news_items [online_users] => online_users [private_files] => private_files [recent_activity] => recent_activity [recentlyaccessedcourses] => recentlyaccessedcourses [recentlyaccesseditems] => recentlyaccesseditems [rss_client] => rss_client [search_forums] => search_forums [section_links] => section_links [selfcompletion] => selfcompletion [settings] => settings [site_main_menu] => site_main_menu [social_activities] => social_activities [starredcourses] => starredcourses [tag_flickr] => tag_flickr [tags] => tags [timeline] => timeline ) 1

So - this is a lot of blocks.  Are you saying that you want to show all of these, and then if any of your 4 favorites are in there, mark them as defaults?

OR, are you wanting me to check and make sure that your 4 favorites are in the list above?  And just show, of the 4, any of those that are also in the above list, thereby making sure that your 4 are in the list of enabled blocks?  And, ditto for defaults?

John




________________________________

**Comment by gjb2048** [comment]
@badmechanic 

- List all of the blocks as 'choices' with the array keys from the list of enabled plugins that you can then use in the 'get_string' as mentioned before.
- Set 'default' to the four as they are, unless one or more is not in 'choices', in that case remove from 'default'.

**Comment by John-Joubert** [comment]
OK - thank you.

________________________________

**Comment by badmechanic** [comment]
OK - well I got the list of enabled blocks,

Now - I want to get the strings from the list.

But when I got after this (for example):
get_string('pluginname','search_forums');
What I get back is:  [[pluginname]]

And it does the same no matter which pluginname I got after.

Of course - I was trying to go after them with a variable where 'search_forums' -
but when it wasn't working - I tried a string for a working block that I know works 'search_forums'.
Just show what it returns.

And - all I get from all of these calls is "[[pluginname]]"

I've looked in the instructions to see if I can find anywhere that get_string can be used to return values from a block, and I cannot.  I find that get_string returns pairs from things defined in the lang files, and that's it.

So, are these strings supposed to be populated in the lang files for us already by the blocks that are installed?

Here's what I see with this from the block...
    /* Toggle display block choices */
    $name = 'format_topcoll/defaultdisplayblocks';
    $title = get_string('defaultdisplayblocks', 'format_topcoll');
    $description = get_string('defaultdisplayblocks_desc', 'format_topcoll');
    $default = array('search_forums', 'news_items', 'calendar_upcoming', 'recent_activity');
//    $choices = array('search_forums' => 'Search forums', 'news_items' => 'Latest announcements',
//                     'calendar_upcoming' => 'Upcoming events', 'recent_activity' => 'Recent activity');
    $choices = core_plugin_manager::instance()->get_enabled_plugins('block');
    foreach ($choices as $key=>$blockname) {
        $choices[$key]=get_string('pluginname',$key);
    }
    die(print_r($choices));

And the output for all the choices in the "die" statement reads:
Array ( [activity_modules] => [[pluginname]] [activity_results] => [[pluginname]] [admin_bookmarks] => [[pluginname]] [badges] => [[pluginname]] [blog_menu] => [[pluginname]] [blog_recent] => [[pluginname]] [blog_tags] => [[pluginname]] [calendar_month] => [[pluginname]] [calendar_upcoming] => [[pluginname]] [comments] => [[pluginname]] [completionstatus] => [[pluginname]] [course_list] => [[pluginname]] [course_summary] => [[pluginname]] [feedback] => Feedback [globalsearch] => [[pluginname]] [glossary_random] => [[pluginname]] [html] => [[pluginname]] [login] => [[pluginname]] [lp] => [[pluginname]] [mentees] => [[pluginname]] [mnet_hosts] => [[pluginname]] [myoverview] => [[pluginname]] [myprofile] => [[pluginname]] [navigation] => [[pluginname]] [news_items] => [[pluginname]] [online_users] => [[pluginname]] [private_files] => [[pluginname]] [recent_activity] => [[pluginname]] [recentlyaccessedcourses] => [[pluginname]] [recentlyaccesseditems] => [[pluginname]] [rss_client] => [[pluginname]] [search_forums] => [[pluginname]] [section_links] => [[pluginname]] [selfcompletion] => [[pluginname]] [settings] => [[pluginname]] [site_main_menu] => [[pluginname]] [social_activities] => [[pluginname]] [starredcourses] => [[pluginname]] [tag_flickr] => [[pluginname]] [tags] => [[pluginname]] [timeline] => [[pluginname]] ) 1


Any direction appreciated.

John
________________________________

**Comment by badmechanic** [comment]
OK - well I found it - I have to get the list of blocks, but to get the string,
I need to prepend "blocks" to the block name in the get_string() function.

So now - this code:
    $choices = core_plugin_manager::instance()->get_enabled_plugins('block');
    foreach ($choices as $key=>$blockname) {
        $choices[$key]=get_string('pluginname','block_' . $key);
    }
    die(print_r($choices));

prints this...
Array ( [activity_modules] => Activities [activity_results] => Activity results [admin_bookmarks] => Admin bookmarks [badges] => Latest badges [blog_menu] => Blog menu [blog_recent] => Recent blog entries [blog_tags] => Blog tags [calendar_month] => Calendar [calendar_upcoming] => Upcoming events [comments] => Comments [completionstatus] => Course completion status [course_list] => Courses [course_summary] => Course/site summary [feedback] => Feedback [globalsearch] => Global search [glossary_random] => Random glossary entry [html] => HTML [login] => Login [lp] => Learning plans [mentees] => Mentees [mnet_hosts] => Network servers [myoverview] => Course overview [myprofile] => Logged in user [navigation] => Navigation [news_items] => Latest announcements [online_users] => Online users [private_files] => Private files [recent_activity] => Recent activity [recentlyaccessedcourses] => Recently accessed courses [recentlyaccesseditems] => Recently accessed items [rss_client] => Remote RSS feeds [search_forums] => Search forums [section_links] => Section links [selfcompletion] => Self completion [settings] => Administration [site_main_menu] => Main menu [social_activities] => Social activities [starredcourses] => Starred courses [tag_flickr] => Flickr [tags] => Tags [timeline] => Timeline ) 1

So - I think I'm on the right track.
I'm not done with my logic - I need to check the above list to see if my defaults are in it, and if so - add that to the defaults list.  But, I'm moving forward on it.

🙂

Lastly ...
Do you have ANY idea what that trailing "1" is at the end of my print_r command against the array?


________________________________

**Comment by gjb2048** [comment]
The '1' will be the exit code from 'die' (https://www.php.net/manual/en/function.die.php), just like 'errno'.

**Comment by gjb2048** [comment]
@badmechanic Thanks, will look when I can.

File: lang/en_us/format_topcoll.php (hunk 7 above)
Comment (gjb2048) [reply lang/en_us/format_topcoll.php]: No longer need: https://github.com/gjb2048/moodle-format_topcoll/pull/98/files#diff-c0088f30815fb5f4137495566dc9e7b9695331b1d54e3338fb13677ab4c55e85R70.

**Comment by gjb2048** [comment]
@badmechanic Looking better, just https://github.com/gjb2048/moodle-format_topcoll/pull/98/files#r638323579 to sort and the Travis issues with settings.php and the coding standard to fix.  I've changed the base of the pull request, so that there will be a way for me to see what happens when I attempt to squash the commits into one on merge for the master branch.

**Comment by badmechanic** [comment]
I'm thinking I don't need to do anything here.
Let me know if I'm wrong, please.

________________________________

**Comment by badmechanic** [comment]
Sorry - this link takes me to something that looks just like the last one.

I see something about code not needed.  But, if I take those elements from the lang files, it's all going to break...

As for travis tool - I don't see it in the links.
I'll go hunting for it.

Also - since you've changed the base of the pull request, and you are collapsing the commits...
I don't know what that means to what I'm doing here.
I don't want to send anything up, as I don't know if it will mess up what you're on, and where you want me to be.
Do I start with a new pull request after you merge?

Thank you, sorry I don't understand the implications ...

________________________________

**Comment by badmechanic** [comment]
OK - I think you're telling me to take the language strings out of en_us, and just use "en".
Is that what you're saying?

If so, I can fix that.

________________________________

**Comment by badmechanic** [comment]
I cannot find a link anywhere to the latest travis results.  I don't see it in my pull on your page...
and I just can't find it.

Can you please send?

________________________________

**Comment by gjb2048** [comment]
@badmechanic RE: "I think you're telling me to take the language strings out of en_us, and just use "en"." = yes, as they are identical to 'en', 'en_us' only needs to contain strings that are different to 'en', i.e. 'colour' -> 'color'.

Travis info here, but just in case -> https://travis-ci.org/github/gjb2048/moodle-format_topcoll/jobs/772129561#L577-L590

**Comment by gjb2048** [comment]
@badmechanic i.e. ->

Screenshot 2021-05-25 125014

**Comment by gjb2048** [comment]
@badmechanic Thanks, but 'en_us' is still showing up in the list of commits with a change -> https://github.com/gjb2048/moodle-format_topcoll/pull/98/files#diff-c0088f30815fb5f4137495566dc9e7b9695331b1d54e3338fb13677ab4c55e85L69 - please make a commit with nothing changed (as Moodle also complains of no blank line at end of file at times).  Then when I squash the commits into one, then hopefully will only be three changed files and you'll retain the commit credit.  I'll see how this goes.

**Comment by badmechanic** [comment]
OK - I did a commit, but it said that my branch is up to date.
Then, I tried a few pushes...

/moodle/course/format/topcoll$ git commit -m "Null Commit"
On branch noblocks_option
Your branch is up to date with 'origin/noblocks_option'.

nothing to commit, working tree clean
/moodle/course/format/topcoll$ git push origin noblocks_option
Everything up-to-date
/moodle/course/format/topcoll$ git push -u origin noblocks_option
Branch 'noblocks_option' set up to track remote branch 'noblocks_option' from 'origin'.
Everything up-to-date

So, since there's no changes - there's no commit, and nothing to push...

Do you want me to make a non-change to a file (just change a word in a comment),
and then retry?

Or- we can just reject everything, and I can Fork the project all over again,
put in the 3 changed files, commit them, and then push it all up again.
Then- you can just abandon everything in this existing fork, if desired.
I don't know if you can just delete the existing fork (or I can) - but if so,
I can just do it all over again and submit one simple branch with three changed files.

John


________________________________

**Comment by gjb2048** [comment]
@badmechanic Ok.... humm, two options:

Get a copy of the older version of the file and put it in your branch, as it's different (as it has the newline), then it would be a candidate for a commit.

or

Do create a new pull branch (leave this one as a backup) and then just commit the three changed files in a new single commit.

As it's a learning process, I'll leave the decision up to you :).

G

**Comment by gjb2048** [comment]
Superseded by #100

**Pull request closed without merging**